_search-plugins/knn/approximate-knn.md

```diff
@@ -253,7 +253,52 @@ POST _bulk
 ...
 ```
 
-After data is ingested, it can be search just like any other `knn_vector` field!
+After data is ingested, it can be searched just like any other `knn_vector` field.
+
+### Additional query parameters
+
+Starting with version 2.16, k-NN plugin supports additional method parameters to tune search results. This can be done by providing the right values in `method_parameters` section of the query. Here is an example of the query
+
+```json
+GET my-knn-index-1/_search
+{
+  "size": 2,
+  "query": {
+    "knn": {
+      "my_vector2": {
+        "vector": [2, 3, 5, 6],
+        "k": 2,
+        "method_parameters" : {
+          "ef_search": 100
+        }
+      }
+    }
+  }
+}
+```
+These parameters are dependent on the combination of engine and method used to build the index. The tables will help with the valid set of parameters. Invalid combination of parameters will throw an error
+
+#### HNSW
+
+`ef_search`
+
+Explores the `ef_search` number of vectors to find top k nearest neighbors. Higher value will help improve recall at the cost of search latency. The value must be greater than 0
+
+Supported engines | Radial query support | Notes
+:--- | :--- | :---
+Nmslib | no | Overrides the value in index settings if present in search query
+Faiss | yes | Overrides the value in index settings if present in search query
+Lucene | no | Engine supports `k` or `ef_search`. The final result can be controlled by `size`. k-NN plugin will pick `max(k, ef_search)`
+
+#### IVF
+
+`nprobes`
+
+Explores `nprobes` number of clusters to find the top k nearest neighbors. Higher value will help improve recall at the cost of search latency. The value must be greater than 0
+
+Supported engines | Notes
+:--- | :--- 
+faiss | Overrides the value in index settings if present in search query
```

Comment:
Please specify the index setting that nprobes overrides in this case.
```suggestion
`faiss` | If `nprobes` is present in a query, it overrides the <which setting> index setting.
```

Reply:
It overrides nprobes value provided [during index request](https://opensearch.org/docs/latest/search-plugins/knn/knn-index/#ivf-parameters)

Reply:
Thanks! Updated the suggestion.